src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -32,27 +32,51 @@
 
 /**
  * Head corresponding to a pull request.
- * Named like {@code PR-123}.
+ * Named like {@code PR-123} or {@code PR-123-merged} or {@code PR-123-unmerged}.
  */
 public final class PullRequestSCMHead extends SCMHead {
 
-    private static final String PR_BRANCH_PREFIX = "PR-";
-
     private static final long serialVersionUID = 1;
 
     private final PullRequestAction metadata;
+    private Boolean merge;
+    private final boolean trusted;
```

Comment:
Defaulting to false for old settings to be on the safe side. Only practical impact should be that PRs filed before the upgrade will start using `Jenkinsfile` from the base branch if you upgrade and then push more commits, even when the PR should have been trusted. I think this is an acceptable price to pay for fixing up the metadata to store this field where it makes more sense, so that we do not need to recheck trusted status on every commit.